python-client/tests/integration/retry_fetch_trace/test_retry_fetch_until.py

```diff
@@ -152,4 +153,18 @@ def num_is_5(trace):
         end = time.time()
         self.assertGreaterEqual(end - start, 10)
         self.assertFalse(response)
-    
+        
+    def test_retry_trace_segment_not_found(self):
+        patched_get_trace_tree = IatkException("found a segment 123456789 with no parent", 500)
+        with patch("aws_iatk.AwsIatk._get_trace_tree", return_value=patched_get_trace_tree):
+            def num_is_5(trace):
+                assert random.randrange(0,5) == 5
+            start = time.time()
+            response = self.iatk.retry_get_trace_tree_until(
+                tracing_header="Root=1-652850da-255d5ae071f55e4aef339837;Sampled=1",
+                assertion_fn=num_is_5,
+                timeout_seconds=10,
+            )
+            end = time.time()
+            self.assertGreaterEqual(end - start, 10)
```

Comment:
nit: Is there a better way to test for the response timeout other than relying on `time.time()`? I see that this is used in other places in this test class too and with the `assertGreaterEqual` method, but was just curious if there was a way to not rely on counting time.

Reply:
Hmm, I tried researching quickly and some decorators for asserting time is less than x for a function, but none for my use case here in taking longer than 10, I am open to any suggestions though